### PR TITLE
✨ Stop if there is no tags in repo and release option is absent

### LINKED ISSUE #14: Notice the user if there is no tag
Print a nice message telling the users their repository needs to have at least one tag unless they use the release option.

### PR DESCRIPTION
**Pull request by charlyx**
Hello @fabienjuif 👋

This is my very first Rust contribution! 🎉

After reading your code I think I've added the message at the right position. Please tell me if I'm wrong!

At first I was going to use `panic!` as I want to display a message and stop the program but the result is not human-friendly.
I could have used [human_panic](https://docs.rs/human-panic/) but I didn't want to introduce a new dependency without talking to you first.

So here I am with a simple solution.

Happy to discuss! 😄 

Fix #14

### CODE REVIEW

**Comment by fabienjuif** [comment]
Nice :)

I read that tomorrow, if I forgot just ping me!

About the new dependencies, I think this is not a panic message since your message is here to help human using the CLI.
So I think you did the right choice!

**Comment by fabienjuif** [comment]
If we wan't the error to be printed as a CLI, we should add it in `main.rs`, line 38 maybe, juste before `changelog.to_markdown(`

But if we wan't the error to be printed as a lib, the message is fine where you put it.
Except that a process.exit() in a lib is not fine :)
We should either:
1. Let the message in CLI only (see first paragraph)
2. Returns a `Result<String, Err> ` in the `to_markdown` method

**Comment by fabienjuif** [comment]
What do you think?, I think the 1. is good enough

**Comment by charlyx** [comment]
Haha! I haven’t thought about the lib to be honest... 😢 

But yeah, I have thought of returning a type Result. Let’s do it. I like it 😊 

I guess we’ll have to deal with the error in the main then?

**Comment by fabienjuif** [comment]
Yes!
Thats it, you can use pattern matching in the main.rs if you want.

https://doc.rust-lang.org/std/result/enum.Result.html

File: src/changelog.rs [hunk 1]
```diff
@@ -89,11 +90,21 @@ impl Changelog {
         Changelog { versions }
     }
 
-    pub fn to_markdown(&self, release: Option<&str>, print_authors: bool) -> String {
+    pub fn to_markdown(
+        &self,
+        release: Option<&str>,
+        print_authors: bool,
+    ) -> Result<String, String> {
```
Comment (fabienjuif) [inline src/changelog.rs]: The overall PR seems ok to me :)

Just here we are used to see `Result<String, Error>` I guess.
Would you do the typed Error on this PR?

File: Cargo.lock [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# This file is automatically @generated by Cargo.
```
Comment (fabienjuif) [inline Cargo.lock:1]: Did you add this manually or is it the latest version of cargo that added that?

Comment (charlyx) [reply Cargo.lock:1]: Cargo did it. I can remove this commit if you like.

`git commit -a` betrayed me 😞

File: src/changelog.rs (hunk 1 above)
Comment (charlyx) [reply src/changelog.rs]: Yep, will do or do you prefer another PR?

I just wanted to make it work and get your approval first.

File: Cargo.lock (hunk 2 above)
Comment (fabienjuif) [reply Cargo.lock:1]: No this is fine, I asked out of curiosity

File: src/changelog.rs (hunk 1 above)
Comment (fabienjuif) [reply src/changelog.rs]: I prefer you do it in this PR then!

File: src/changelog.rs [hunk 3]
```diff
@@ -89,11 +91,17 @@ impl Changelog {
         Changelog { versions }
     }
 
-    pub fn to_markdown(&self, release: Option<&str>, print_authors: bool) -> String {
+    pub fn to_markdown(&self, release: Option<&str>, print_authors: bool) -> Result<String, Error> {
```
Comment (fabienjuif) [inline src/changelog.rs:94]: Is there a way to expose the std::Error instead of our custom one?

File: src/main.rs [hunk 4]
```diff
@@ -35,10 +37,16 @@ fn main() {
...
+        match changelog.to_markdown(
             matches.value_of("release"),
             matches.is_present("print-authors"),
-        )
+        ) {
+            Err(error) => {
+                eprintln!("{}", error);
+                process::exit(1);
+            }
+            Ok(delta) => delta,
```
Comment (fabienjuif) [inline src/main.rs:48]: You can also do
`_ => _` I think

Comment (charlyx) [reply src/main.rs:48]: Nope: 

```
error: expected expression, found reserved identifier `_`
  --> src/main.rs:48:18
   |
48 |             _ => _
   |               -- ^ expected expression
   |               |
   |               while parsing the `match` arm starting here
```

**Comment by fabienjuif** [comment]
Thank you master, I publish it right now in 0.4.0

**Pull request merged**